File: 22-秋招/笔试/220831-广联达/2.c
#include <stdio.h>

int main()
{
  int n, m, ans = 0;
  scanf("%d%d", &n, &m);
  int flag[n+1];
  for (int i = 0; i <= n; ++i)
  {
    flag[i] = 1;
  }
  flag[0] = 0;
  for (int i = 0; i < m; ++i)
  {
    int left, right, x;
    scanf("%d%d%d", &left, &right, &x);
    int temp = x;
    for (int j = left; j <= right; ++j)
    {
      if (temp > 0)
      {
        if (flag[j])
        {
          temp--;
        }
      }
      else
      {
        flag[j] = 0;
      }
    }
  }
  for (int i = 0; i <= n; ++i)
  {
    ans += flag[i];
  }
  printf("%d\n", ans);
  return 0;
}

File: 22-秋招/笔试/220831-广联达/2.js
function solution(n = 10, m = 3, nums = []) {
  nums.sort((a, b) => (a[0] === b[0] ? a[1] - b[1] : a[0] - b[0]));
  let ans = 0;
  let flag = new Array(n).fill(1);

  for (const [left, right, x] of nums) {
    let temp = 0;
    for (let i = left - 1; i <= right - 1; ++i) {
      if (temp === x) {
        flag[i] = 0;
        continue;
      }
      if (flag[i] && temp < x) {
        temp += 1;
      }
    }
  }

  flag.forEach((f) => {
    ans += f;
  });

  return ans;
}

function main() {
  const n = readInt(),
    m = readInt();
  const nums = [];
  for (let i = 0; i < m; ++i) {
    const arr = [];
    arr.push(readInt());
    arr.push(readInt());
    arr.push(readInt());
    nums.push(arr);
  }

  console.log(solution(n, m, nums));
}

main();
